include/dos_windows.h

```diff
@@ -0,0 +1,143 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2025-2025  The DOSBox Staging Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#ifndef DOSBOX_DOS_WINDOWS_H
+#define DOSBOX_DOS_WINDOWS_H
+
+#include <cstdint>
+#include <utility>
+
+#include "compiler.h"
+#include "mem.h"
+
+// Functions and definitions which can be used by programs and drivers to
+// communicate with Microsoft Windows running on the simulated DOS.
+
+// WARNING: Windows releases before 3.1 do not support all the APIs listed here,
+// in particular Windows 1.0 does not support any of the calls.
+
+using WindowsVmId = uint16_t;
+
+// Virtual Machine ID used by the Windows kernel
+constexpr WindowsVmId WindowsKernelVmId = 1;
+
+// If Microsoft Windows is running (Windows 1.0 won't be detected).
+bool WINDOWS_IsStarted();
+
+// Get the major/minor version of the currently running Windows
+std::pair<uint8_t, uint8_t> WINDOWS_GetVersion();
+
+// Check if Windows is running in the 386 Enhanced mode
+bool WINDOWS_IsEnhancedMode();
+
+// Get address of the 386 Enhanced mode Windows entry point procedure. Only
+// intended to be used under Windows 2.x, under Windows 3.x it is provided
+// for compatibility with earlier software
+std::pair<uint16_t, uint16_t> WINDOWS_GetEnhancedModeEntryPoint();
```

Comment:
I don't really like pairs, they don't document what the two values are. Consider using a struct instead.